consensus/src/consensus/mod.rs

```diff
@@ -662,8 +663,8 @@ impl ConsensusApi for Consensus {
         self.coinbase_manager.modify_coinbase_payload(payload, miner_data)
     }
 
-    fn validate_pruning_proof(self: Arc<Self>, _proof: &PruningPointProof) -> Result<(), PruningImportError> {
-        unimplemented!()
+    fn validate_pruning_proof(self: Arc<Self>, proof: &PruningPointProof) -> Result<(), PruningImportError> {
```

Comment:
change the interface and avoid the `Arc<Self>` and use `&self` (now all consensus api methods are like this)